interop/ksp/src/main/kotlin/com/squareup/kotlinpoet/ksp/Annotations.kt

```diff
@@ -39,19 +39,35 @@ public fun KSAnnotation.toAnnotationSpec(omitDefaultValues: Boolean = false): An
     is ParameterizedTypeName -> AnnotationSpec.builder(type)
     else -> error("This is never possible.")
   }
+  val params = (annotationType.resolve().declaration as KSClassDeclaration).primaryConstructor?.parameters.orEmpty()
+    .associateBy { it.name }
   useSiteTarget?.let { builder.useSiteTarget(it.kpAnalog) }
   // TODO support type params once they're exposed https://github.com/google/ksp/issues/753
+  var varargValues: List<*>? = null
   for (argument in arguments) {
     val value = argument.value ?: continue
     val name = argument.name!!.getShortName()
+    val type = params[argument.name]
     if (omitDefaultValues) {
       val defaultValue = this.defaultArguments.firstOrNull { it.name?.asString() == name }?.value
       if (isDefaultValue(value, defaultValue)) { continue }
     }
-    val member = CodeBlock.builder()
-    member.add("%N = ", name)
-    addValueToBlock(value, member, omitDefaultValues)
-    builder.addMember(member.build())
+    if (type?.isVararg == true) {
+      // wait to add varargs to end
```

Comment:
super nit: [comments should be sentences](https://nedbatchelder.com/blog/201401/comments_should_be_sentences.html)
```suggestion
      // Wait to add varargs to end.
```